src/NuGet.Core/NuGet.Common/NuGetEventSource.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved. Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// Represents a class used for logging trace events from NuGet.
+    /// </summary>
+    public static class NuGetEventSource
+    {
+        /// <summary>
+        /// Gets a <see cref="NuGetEventSource" /> which can be used to trace events from NuGet.
+        /// </summary>
+        public static EventSource Instance { get; } = new EventSource("Microsoft-NuGet");
+
+        /// <summary>
+        /// Represents a class for declaring event keywords. Each keyword must be a flag (2^N) for use in a bitwise operation.
+        /// </summary>
+        public static class Keywords
+        {
+            /// <summary>
+            /// The event keyword for tracing events related to NuGet's common functionality from the NuGet.Common namespace.
+            /// </summary>
+            public const EventKeywords Common = (EventKeywords)1;
```

Comment:
So guessing this in response to @zivkan's suggestion that we have differentiators for each assembly?

Reply:
This doesn't quite "differentiate" the events but it does let you filter out events you might not care about depending on the scenario.  I have no idea if we'll ever use them but I can imagine if we have a ton of events then we might ask a customer for only NuGetSdkResolver events for example.